src/Npgsql/NpgsqlReadBuffer.cs

```diff
@@ -41,20 +41,17 @@ internal TimeSpan Timeout
             get => Cts.Timeout;
             set
             {
+                if (value == TimeSpan.Zero)
```

Comment:
Here again, for perf I'd rather not to these conditions all the time if the timeout doesn't change. We can do that by caching the "pre-translated" timeout value in a field, and comparing against that. If it doesn't match, only then we translate 0->infinite, negative->zero, etc.

Reply:
The main thing to remember, is that we should always optimize for a program running normally, without either cancellations or timeouts happening - this is the path we should keep very fast. The assumption is also that users almost never change timeouts between different commands - it's just something they set on the connection string and it never changes.